app/Console/Commands/InteractsWithCollections.php

```diff
@@ -30,12 +30,21 @@ public function forEachCollection(Closure $callback, Closure $queryCallback = nu
             return;
         }
 
-        Collection::query()
+        $query = Collection::query()
             ->when($queryCallback !== null, $queryCallback)
             ->select('collections.*')
             ->withoutSpamContracts()
-            ->chunkById(100, function ($collections) use ($callback) {
-                $collections->each($callback);
-            }, 'collections.id', 'id');
+            ->when($limit !== null, fn ($query) => $query
+                ->limit($limit)
+                ->get()
+                ->filter(fn ($collection) => ! $collection->isBlacklisted())
+                ->each($callback)
+            )
+            ->when($limit == null, fn ($query) => $query->chunkById(100, function ($collections) use ($callback) {
+                $collections
+                    ->filter(fn ($collection) => ! $collection->isBlacklisted())
```

Comment:
This implementation breaks [these](https://app.clickup.com/t/85zu0hdha) remarks:

> spam => ignore for fetching NFTs and metadata / banner
> supply null/too large => ignore for fetching NFTs
> blacklist => ignore for fetching NFTs

The `InteractsWithCollections` is being used by many jobs and not all of them should filter blacklisted collections, since the default value of `limit=null` in the function signature this will cause unexpected results.